pallets/omnipool/src/lib.rs

```diff
@@ -2017,18 +2124,48 @@ impl<T: Config> Pallet<T> {
 	}
 
 	/// Calls `on_trade_fee` hook and ensures that no more than the fee amount is transferred.
-	fn process_trade_fee(trader: &T::AccountId, asset: T::AssetId, amount: Balance) -> DispatchResult {
+	fn process_trade_fee(
+		trader: &T::AccountId,
+		asset: T::AssetId,
+		amount: Balance,
+	) -> Result<Vec<Fee<T::AccountId>>, DispatchError> {
 		let account = Self::protocol_account();
 		let original_asset_reserve = T::Currency::free_balance(asset, &account);
 
 		// Let's give little bit less to process. Subtracting one due to potential rounding errors
 		let allowed_amount = amount.saturating_sub(Balance::one());
-		let used = T::OmnipoolHooks::on_trade_fee(account.clone(), trader.clone(), asset, allowed_amount)?;
+		let fees = T::OmnipoolHooks::on_trade_fee(account.clone(), trader.clone(), asset, allowed_amount)?;
+		let additional_fees_total: Balance = fees
+			.clone()
+			.into_iter()
+			.filter_map(|opt| opt.map(|(balance, _)| balance))
```

Comment:
since you filter out zero amounts later on, you can do it directly here

```suggestion
			.filter_map(|opt| opt.map(|(balance, _)| if balance.is_zero(){
			     None
			}
			else{
			    Some(balance)
			}))
```

something like that should work

Reply:
but that should not really return 0 fee amounts anyway.

Reply:
i see your point, but here I sum the total, instead of converting them to some entries, so having the current implementation makes more sense imo.